pageserver/src/walingest.rs

```diff
@@ -1049,16 +1049,18 @@ impl WalIngest {
 
         // Delete all the segments except the last one. The last segment can still
         // contain, possibly partially, valid data.
-        while segment != endsegment {
-            modification
-                .drop_slru_segment(SlruKind::MultiXactMembers, segment as u32, ctx)
-                .await?;
+        if modification.tline.get_shard_identity().is_shard_zero() {
```

Comment:
Shouldn't this check be at the start of the method? And why are these records even sent to other shards?

Reply:
> Shouldn't this check be at the start of the method?

No, each shard needs to update it's checkpoint.

> And why are these records even sent to other shards?

Checkpoint updates, I think.